Svc/ActiveLogger/ActiveLoggerImpl.cpp

```diff
@@ -100,8 +100,10 @@ namespace Svc {
             }
         }
 
+        Fw::LogSeverity severity_copy = severity;
+
         // send event to the logger thread
-        this->loqQueue_internalInterfaceInvoke(id,timeTag,severity,args);
+        this->loqQueue_internalInterfaceInvoke(id,timeTag,severity_copy,args);
```

Comment:
This inconsistency is woven into the autocoder, via similar-but-not-quite-the-same code for handling ports and internal interfaces. The current PR does not change the behavior for internal interfaces.

I think the current behavior for internal interfaces is acceptable. It ignores the `pass_by` attribute and unconditionally passes primitive types by value and non-primitive types by (mutable) reference.

I recommend that we accept this behavior for release 3.0 and revisit later, when we revise the entire C++ back end.

Reply:
I removed `pass_by="reference"` from the internal interfaces of the active loggers, because it does nothing.

Reply:
However, I don't know why `severity_copy` was added there. I will investigate.

Reply:
Actually, now that I look at the diff in ActiveLogger, it seems that before this PR the autocoder was sensitive to `pass_by="reference"`. I'll have to take another look at the autocoder code.

Reply:
I guess `severity_copy` was added in order to pass the data from a const ref into a non-const ref. As I mentioned, we could fix this but it would take some work. I think it's out of scope for this PR.

Reply:
Possibly we could revise the internal interface code to use a const ref instead of a non-const ref. Maybe we could do this without major changes. I will investigate.

Reply:
Confirmed, I improved the const correctness and removed the extra copy. Thanks, @LeStarch!